src/classes/Contacts.cls

```diff
@@ -491,6 +625,135 @@ public inherited sharing class Contacts extends fflib_SObjects {
+            if (contactRecord.AccountId != null && contactRecord.npe01__Private__c == true) {
+                contactRecord.AccountId = null;
+            }
+        }
+    }
+
+    public Set<Id> oldAccountIds() {
+        Set<Id> accountIds = new Set<Id>();
+        for (Contact contactRecord : oldMap.values()) {
+            accountIds.add(contactRecord.AccountId);
+        }
+        return accountIds;
+    }
+
+    public Map<Id, Account> accountById {
+        get {
+            if (accountById == null) {
+                loadAccountByIdMap();
+            }
+            return accountById;
+        }
+        set;
+    }
+
+    public List<Contact> contactsWithoutAccounts() {
+        List<Contact> contactsWithoutAccounts = new List<Contact>();
+        for (Contact contactRecord : (List<Contact>) getRecords()) {
+            if (isMissingAccount(contactRecord)) {
+                contactsWithoutAccounts.add(contactRecord);
+            }
+        }
+        return contactsWithoutAccounts;
+    }
+
+    public Map<Id, Contact> getNewOwnerContactById() {
+        Map<Id, Contact> newOwnerContactById = new Map<Id, Contact>();
+        for (Contact contactRecord : (List<Contact>) getRecords()) {
+            Contact oldContact = oldMap.get(contactRecord.Id);
+            if (contactRecord.OwnerId != oldContact.OwnerId) {
+                newOwnerContactById.put(contactRecord.Id, contactRecord);
+            }
+        }
+        return newOwnerContactById;
+    }
+
+    /**
+     * @description Determines whether or not the contact requires a household account.
+     * @param  contactRecord The contact to verify.
+     */
+    public Boolean needsHHAccount(Contact contactRecord) {
+        return CAO_Constants.isHHAccountModel()
+                && contactRecord.AccountId == null
+                && contactRecord.npe01__Private__c != true;
+    }
+
+    /*******************************************************************************************************
+    * @description returns whether changes in the contact require an update to the 1:1 account
+    * @param contactRecord the new version of the Contact
+    * @param oldContact the old version of the Contact
+    * @return Boolean
+    */
+    private Boolean needsOne2OneAccountUpdate(Contact contactRecord, Contact oldContact) {
+        if (contactRecord.npe01__Organization_Type__c != CAO_Constants.ONE_TO_ONE_ORGANIZATION_TYPE) {
+            return false;
+        }
+
+        if(contactRecord != null && oldContact == null) {
+            return true;
+        }
+
+        String newFirst = contactRecord.FirstName == null ? '' : contactRecord.FirstName;
+        String oldFirst = oldContact.FirstName == null ? '' : oldContact.FirstName;
+        String newLast = contactRecord.LastName == null ? '' : contactRecord.LastName;
+        String oldLast = oldContact.LastName == null ? '' : oldContact.LastName;
+
+        return (contactRecord.AccountId != oldContact.AccountId ||
```

Comment:
Out of scope if we're just trying to reorganize, but this could just be a loop over a list of field names.